### PR TITLE
reference: vtable entries are resolved at runtime

### PR DESCRIPTION
**Pull request by ivan**
I sure hope this fix is right

cc @steveklabnik

### CODE REVIEW

**Comment by jonas-schievink** [comment]
I think the reference was right, but perhaps a bit unclear: The vtable itself *is* built at compile time, so the vtable entry for all methods is known at compile time, but it's not known which method the entry actually points to.

**Comment by aturon** [comment]
I agree with @jonas-schievink. It'd be mildly more clear as 

> a method on a trait object is resolved only to a vtable entry at compile time

instead of:

> a method on a trait object is only resolved to a vtable entry at compile time

but, honestly, the whole paragraph could probably use a rewrite for clarity.

File: src/doc/reference.md [hunk 1]
```diff
@@ -3565,7 +3565,7 @@ Each instance of a trait object includes:
    implementation (i.e. a function pointer).
 
 The purpose of trait objects is to permit "late binding" of methods. A call to
-a method on a trait object is only resolved to a vtable entry at compile time.
+a method on a trait object is only resolved to a vtable entry at runtime.
```
Comment (nikomatsakis) [inline src/doc/reference.md]: I think I would probably say something like "Calling a method on a trait object results in virtual dispatch at runtime: that is, a function pointer is loaded from the trait object vtable and invoked indirectly."

**Pull request merged**